src/pages/signin/SAMLSignInPage/index.native.js

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React, {useCallback, useState} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import WebView from 'react-native-webview';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import SAMLLoadingIndicator from '@components/SAMLLoadingIndicator';
+import ScreenWrapper from '@components/ScreenWrapper';
+import getPlatform from '@libs/getPlatform';
+import Navigation from '@libs/Navigation/Navigation';
+import * as Session from '@userActions/Session';
+import CONFIG from '@src/CONFIG';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+
+const propTypes = {
+    /** The credentials of the logged in person */
+    credentials: PropTypes.shape({
+        /** The email/phone the user logged in with */
+        login: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    credentials: {},
+};
+
+function SAMLSignInPage({credentials}) {
+    const samlLoginURL = `${CONFIG.EXPENSIFY.SAML_URL}?email=${credentials.login}&referer=${CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER}&platform=${getPlatform()}`;
+    const [showNavigation, shouldShowNavigation] = useState(true);
+
+    /**
+     * Handles in-app navigation once we get a response back from Expensify
+     *
+     * @param {String} params.url
+     */
+    const handleNavigationStateChange = useCallback(
+        ({url}) => {
+            // If we've gotten a callback then remove the option to navigate back to the sign in page
+            if (url.includes('loginCallback')) {
+                shouldShowNavigation(false);
+            }
+
+            const searchParams = new URLSearchParams(new URL(url).search);
+            if (searchParams.has('shortLivedAuthToken')) {
+                const shortLivedAuthToken = searchParams.get('shortLivedAuthToken');
+                Session.signInWithShortLivedAuthToken(credentials.login, shortLivedAuthToken);
+            }
+
+            if (searchParams.has('error')) {
+                // Run the Onyx action to set an error state on the sign in page
+                // Currently this is what's going to trigger because the backend isn't redirecting SAML correctly
+                Navigation.navigate(ROUTES.HOME);
+            }
```

Comment:
Ah yes, you're right @fedirjh - I thought it might be how we're handling the error being thrown in the backend, but really we just need to set the account error here 👍